semgrep_output_v1.atd

```diff
@@ -1680,6 +1686,42 @@ type ci_scan_complete_stats = {
     <python repr="dict">
     <ts repr="map">
     option;
+
+  (* since 1.98.0 *)
+  (* In collaboration with the Data Science team, it was suggested
+   * that we start to group stats by product for organizational purposes.
+   *)
+  ?supply_chain_stats: supply_chain_stats option;
+}
+
+(* Valid schema versions. Update when making breaking changes. *)
+type stats_schema_version <ocaml attr="deriving show"> <python decorator="dataclass(frozen=True)"> = [
+  | V1_0_0 <json name="1.0.0"> (* initial version *)
+] 
+
+type resolution_method <ocaml attr="deriving show"> <python decorator="dataclass(frozen=True, order=True)"> = [
+  | LockfileParsing 
+  | DynamicResolution 
+]
+
+type dependency_resolution_stats = {
+  resolution_method: resolution_method;
+  dependency_count: int;
+  ecosystem: ecosystem;
+}
+
+type subproject_stats = {
+  dependency_source_files: string list;
+  (* Results of dependency resolution, null if resolution failed *)
+  resolved_stats: dependency_resolution_stats nullable;
```

Comment:
why nullable here instead of option?

Reply:
No explicit reason. I was working of this sample json:
```
{
    "subproject_stats": [
        ...
        {
            "dependency_sources": ["proj_c/gradle.kts"],
            "resolved_stats": null
        }
    ]
}
```

In this case, nullable makes it clear that `resolved_stats` explicity has no value. In the end, i don't think it has impact on the downstream metrics -- just on how we use it!

Reply:
Ah, looks like `nullable` in ATD still becomes an `option` in Ocaml - seems solid then! https://atd.readthedocs.io/en/latest/atdgen.html#default-type-mapping

Reply:
Perfect! I'll leave as is then!